### ISSUE TITLE
Require numpy for ITK Python wrapping

### ISSUE THREAD

**Issue by hjmjohnson** [InsightSoftwareConsortium/ITK#2055]
### Description

There is a large amount of conditional code to support python wrapping of ITK without numpy.  This appears to not be tested well, and it greatly complicates understanding the python code.

### Impact analysis

Maintainance of ITK python wrapping is significantly complicated by attempting to make the wrapping work both with and without numpy.

### Expected behavior
A balance of benefits maintenance burden should be achieved.

### Actual behavior

The maintenance burden of python wrapping for the (assumed) minimal benefits of using ITK without numpy are not justified.

### Versions
All previous wrapped versions of ITK python wrapping.

### Additional Information

I propose that numpy becomes a mandatory pre-requisite for ITK python wrapping.

**Comment by dzenanz** [comment]
I think that we decided than `numpy` is mandatory some time ago. But maybe not all the code has been cleaned up. @thewtex is that right?

**Comment by jhlegarreta** [comment]
I am not sure about a few aspects concerning this, since I have not been involved in programming the support itself:
- What would be an example of a feature that would no longer be available if the `NumPy`-free support was removed? e.g. like a call or few lines of code?
- I ignore the workload involved in maintaining `NumPy`-free wrapping compatibility.
- I ignore the short-term workload involved in removing the `NumPy`-free wrapping compatibility (e.g. like the files and classes that would be affected by the change).
- I do not know what is the adoption of the `NumPy`-free wrapping among the `ITK` community.

When I use `Python` outside `ITK`, the first thing I do is to import `NumPy`, almost always. Not sure about its future, but as of today, I think it is the package of choice to manipulate almost anything in scientific computing.

Sorry to cast more uncertainty rather than giving answers, but I feel that I cannot provide a meaningful opinion on this.

**Comment by thewtex** [comment]
We removed NumPy as a build-time dependency a while ago, which was helpful for build setup.

For the run-time binary Python package, `numpy` is a depedency.

We have avoided adding `numpy` as a testing dependency -- this has been nice because there are no extra Python package dependencies to build and test. NumPy-related tests are skipped unless NumPy is available. However, we have dramatically improved NumPy integration recently, and as @hjmjohnson suggests, it is time to add it as a test-time dependency.

The next steps are to add `numpy` installation to the GitHub CI configurations. `numpy` is now pip-installable across platforms.
